src/components/checkbox.mbt

```diff
@@ -0,0 +1,141 @@
+///|
+/// Creates a configurable checkbox component with customizable styling and
+/// behavior.
+///
+/// Parameters:
+///
+/// * `checked` : Bool - Whether the checkbox is checked.
+/// * `on_change` : M - The message or action to be dispatched when the checkbox
+///   state changes.
+/// * `label` : String? - Optional label text to display next to the checkbox.
+/// * `size` : @theme.CheckboxSize - The size of the checkbox. Defaults to
+///   `@theme.Md`.
+/// * `color` : @theme.CheckboxColor - The color theme of the checkbox. Defaults to
+///   `@theme.Primary`.
+/// * `disabled` : Bool - Whether the checkbox is disabled. Defaults to `false`.
+/// * `className` : String? - Additional CSS class name to apply to the checkbox.
+///   Defaults to `None`.
+///
+/// Returns an HTML checkbox element configured with the specified properties and
+/// styling.
+///
+/// Example:
+///
+/// ```moonbit
+/// let my_checkbox = @jade.checkbox(
+///   checked=true,
+///   on_change=MyCheckboxMessage,
+///   label=Some("Accept terms"),
+///   size=Lg,
+///   color=Success
+/// )
+/// ```
+///
+pub fn[M] checkbox(
+  checked : Bool,
+  on_change : M,
+  id? : String,
+  name? : String,
+  label~ : String? = None,
+  size~ : @theme.CheckboxSize = @theme.Md,
+  color~ : @theme.CheckboxColor = @theme.Primary,
+  disabled~ : Bool = false,
+  className~ : String? = None
```

Comment:
The `class` is not a keyword in MoonBit; you can use `class` as the parameter name.